lymph/node.py

```diff
@@ -123,21 +124,15 @@ def __init__(self, name: str, state: int) -> None:
 
         A tumor can only ever be in one state, and it cannot change its state.
         """
-        allowed_states = [state]
-        super().__init__(name, state, allowed_states)
+        self.allowed_states = [state]
+        super().__init__(name, state, self.allowed_states)

```

Comment:
Now you're setting an attribute and then pass that attribute to its own constructor to set it again...

Was there an issue with the original implementation? One could make it shorter (but less clear) by just writing
```python
super().__init__(name, state, [state])
```
But I'm not a huge fan of that either.

Reply:
allowed_states was not defined before that, so we need to give a list of allowed states. Since the `super.__init__`  does:

```
if allowed_states is None:
            allowed_states = [0, 1]
```
we would give the tumor node the wrong states, hence I go with your proposition

Reply:
But `allowed_states` is _not_ `None` when we initialize it. In the `Tumor` constructor I explicitly pass the `allowed_states = [state]` to the `__init__` method of `AbstractNode`.